### PR TITLE
Autocomplete and Input

### PR DESCRIPTION
**Pull request by AlleeX**
### WHAT DOES THIS PR DO: ###

Adds the Autocomplete and Input component to Travix UI Kit.

selection_247

selection_249

selection_246

selection_248

### WHERE SHOULD THE REVIEWER START: ###
Diffs

### UNIT AND/OR FUNCTIONAL TESTS: ###

Adds Unit tests for Autocomplete and Input component

### CODE REVIEW

File: components/autoComplete/autoComplete.scss [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  box-shadow: 0 0 5px 0 rgba($tx-autocomplete-box-shadow, .18);
+  left: 0;
+  margin-top: -2px;
+  max-height: 300px;
+  overflow-x: overlay;
+  position: absolute;
+  right: 0;
+  top: 100%;
+  z-index: 3;
+  display: none;
```
Comment (AlexDudar) [inline components/autoComplete/autoComplete.scss]: could you please sort alphabetically

File: components/autoComplete/autoComplete.js [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+
+    const item = this.items.length && this.items[activeKey].getValue();
+
+    if (!item || (!this.state.inputValue && e.keyCode !== KEY_CODE.ENTER)) {
+      this.setState({
+        inputValue: undefined,
+        selectedValue: undefined,
+        open: false,
+      }, () => {
+        this.updateInput(undefined);
```
Comment (asci) [inline components/autoComplete/autoComplete.js]: why do you need to pass `undefined`?

File: components/autoComplete/autoComplete.js [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+  PropTypes,
+} from 'react';
+
+import AutoCompleteItem from './autoCompleteItem';
+import Input from '../input/input';
+import KEY_CODE from '../constants/keyCode';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+
+const getNextKey = (keys, key) => {
+  return keys[keys.indexOf(key) + 1] || key;
```
Comment (asci) [inline components/autoComplete/autoComplete.js]: do you need to have parent context here?

Comment (AlleeX) [inline components/autoComplete/autoComplete.js]: Done, thanks for catch)

File: components/autoComplete/autoComplete.js (hunk 2 above)
Comment (AlleeX) [inline components/autoComplete/autoComplete.js]: improved

File: components/autoComplete/autoComplete.scss (hunk 1 above)
Comment (AlleeX) [inline components/autoComplete/autoComplete.scss]: done

**Pull request merged**